docs/misc/commands.md

```diff
@@ -0,0 +1,211 @@
+```java
+public Dynamic2CommandExceptionType floatTooHigh() {
+    return FLOAT_TOO_BIG;
+}
+```
+
+:::tip
+These functions are accessible everywhere, and it's recommended to use it when throwing `CommandSyntaxException`.
+```java
+throw new CommandSyntaxException(BuiltInException.floatTooHigh(), new LiteralMessage("Please provide a smaller float!"));
+```
+:::
+
+### `SharedSuggestionProvider`
+It's the base interface of execution contexts, [client-side or server-side](../concepts/sides.md).
+
+:::warning
+We talk here about **execution** context. It's meaning that the command hasn't be written when it's instanciated, so you can't get arguments from it, use instead [`CommandContext`](#command-contexts).
+:::
+
+#### `ClientSuggestionProvider`
+It implement `SharedSuggestionProvider` but have many more function, about client context.
+
+### Command Source Stack
+The `CommandSourceStack' holds all the information about the command execution context **before** the command is written and executed.
+
+The most important ones are
+ - `boolean #hasPermission(int)`: returns whether the command executor has the given [permission level](https://minecraft.wiki/w/Permission_level).
+ - `void #sendSuccess(Supplier<Component>, boolean)`: send a success message to the executor. The supplier provides the message and the boolean if it's visible to everyone.
+ - `void #sendFailure(Component)`: send a red failure message to the executor.
+
+#### `ClientCommandSourceStack`
+It's a Neoforge class that implements `SharedSuggestionProvider`. It has no other functions, so it works like the base context.
+
+### The `Commands` class
+It's the "util" class of Minecraft's commands. You can use some of its function for yours, like `Commands.argument`.
+It also work as the register of all Minecraft's and Neoforge's commands, but you can't use it without [Mixin](https://spongepowered.org) (not recommended).
+
+## Creating a command
+To create a command, you'll need to register it with `RegisterCommandsEvent` [event](../concepts/events.md). Here's an example:
+```java
+// Assuming we already regsiter the event
+@SubscribeEvent // Default event bus
+public static void registerCommands(RegisterCommandsEvent event) {
+    event.getDispatcher().register( // event.getDispatcher returns the CommandDispatcher
+        Commands.literal("mycommand") // Always start with this, the name of your command
+            .requires(source -> source.hasPermission(2)) // Check permissions level
+            .then(Commands.argument("level", StringArgumentType.word())) // Here we create an on-word argument named "level"
+            .executes(context -> {
+                if(context.getEntity() instanceof Player player) {
+                    player.sendServerMessage(Component.literal("You're now on the level: " + 
+                        StringArgumentType.getString(context, "level"))); // And we send a message to the executor
+                }
+                return Command.SINGLE_SUCCESS;
```

Comment:
Why success if the command doesn't do anything if its not from a player? It would be more useful to return an error message.